src/helpers/components.js

```diff
@@ -17,18 +15,32 @@ exports.extractComponentOptions = (path) => {
   }
 
   const script = Component.script.content
-  const parsed = acorn.Parser.extend(dynamicImport.default).parse(script, {
-    ecmaVersion: 10,
-    sourceType: 'module'
+  const parsed = parser.parse(script, {
+    sourceType: 'module',
+    plugins: [
+      'classProperties',
+      'decorators-legacy',
+      'dynamicImport',
+      'estree',
+      'exportDefaultFrom',
+      'typescript'
+    ]
   })
-  walker.simple(parsed, {
-    Property (node) {
-      if (node.key.name === COMPONENT_OPTIONS_KEY) {
-        const data = script.substring(node.start, node.end)
-        componentOptions = eval(`({${data}})`)[COMPONENT_OPTIONS_KEY] // eslint-disable-line
+
+  traverse(parsed, {
+    enter (path) {
+      if (path.node.type === 'Property') {
+        if (path.node.key.name === COMPONENT_OPTIONS_KEY) {
+          const data = script.substring(path.node.start, path.node.end)
+          try {
+            componentOptions = eval(`({${data}})`)[COMPONENT_OPTIONS_KEY] // eslint-disable-line
+          } catch (error) {
+            // Ignore
```

Comment:
Should we log the error here? If anything breaks here, it would be useful for user to have a message they can paste when reporting the error.

Reply:
I'll add a warning.

But note that this is only catching `eval` of data, not parsing of the vue file itself.

Data can be wrong in two ways:
 - component option has JS syntax error - it will be caught by Nuxt later, when parsing same page. It's better to let Nuxt fail as it gives much better error and stacktrace than `eval` failing.
 - Code is formed correctly but not in the form that nuxt-i18n expects and thus won't eval successfully. For that case Nuxt will not fail at parsing the the vue file.

So yes, even though for the first case we wouldn't need a warning, I'll add it for both anyways (it's not that I can distinguish them easily anyway).

Reply:
I've actually expanded try/catch to parsing also as that is the one that gives bad errors and Nuxt should report better one later anyway.

Errors when `eval`'ing or using Function constructor are not really catchable but try/catch wraps that part too just in case.